### PR TITLE
Refactor and rename files to be more descriptive

### PR DESCRIPTION
**Pull request by devin-petersohn**
* Rename `*BlockPartitions` to `*FramePartitionManager`
* Rename `*AxisPartition` to `*FrameFullAxisPartition`
* Rename `*RemotePartition` to `*FramePartition`
* Add a `frame` submodule for each engine (including experimental)
  * Add all of the current files to the `frame` submodules for each
    engine
* Add a `series` submodule for each engine
* Add `generic` to `modin.engines.ray` for functionality common to all backends running on Ray
  * Add `io.py` to `modin.engines.ray.generic` for the reading functionality
  * Add `partition_manager.py` to `modin.engines.ray.generic.frame` for `FrameManager` functionality

## Related issue number

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [ ] tests added and passing

### CODE REVIEW

**Comment by devin-petersohn** [comment]
cc @pcmoritz

File: modin/engines/base/README.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 ### Object Hierarchy
 
-- `remote_partition.py` contains `BaseRemotePartition` interface and its implementations.
-- `partition_collections.py` contains `BaseBlockPartitions` interface and its implementations.
-	- `BaseBlockPartitions` manages 2D-array of `BaseRemotePartition` object
-- `axis_partition.py` contains `BaseAxisPartition` and with the following hierarchy:
+- `remote_partition.py` contains `BaseFramePartition` interface and its implementations.
```
Comment (williamma12) [inline modin/engines/base/README.md]: Should we update the file names as well?

File: modin/engines/base/frame/axis_partition.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from modin.data_management.utils import split_result_of_axis_func_pandas
 
 
-class BaseAxisPartition(object):  # pragma: no cover
+class BaseFrameFullAxisPartition(object):  # pragma: no cover
```
Comment (williamma12) [inline modin/engines/base/frame/axis_partition.py]: Why not just `BaseFrameAxisPartition`?

**Comment by pcmoritz** [comment]
Some of the names are getting very long. I'll make some suggestions, but they should be taken with a few grains of salt right now since I don't know the code well yet.

Instead of `FramePartitionManager`, have you considered just `FrameManager` (and `SeriesManager`)? Given these are all in the modin namespace, it still seems quite clear that it is distributed (also indicated by the name "manager").

Why the "Full" in "FullAxisPartition"? I'd consider just dropping it.

I like the change `RemotePartition` to `FramePartition`.

Instead of `PandasOnX`, just call it `Modin`.

In general, I think it would be good to avoid names that are longer than 3 nouns. I'll also reserve the right to change my mind on these ;)

**Comment by devin-petersohn** [comment]
Thanks @pcmoritz. The `PandasOnX` is for pandas backends only. Calling it `Modin` would remove that identity. We have `PandasOnRay` and `PandasOnDask` and `PandasOnPython`. Maybe just `PandasAxisPartition`?

**Comment by williamma12** [comment]
Perhaps the `PandasOnRay`, etc. may not be necessary because we reference it in the import statement e.g., https://github.com/modin-project/modin/blob/6e24ef779c0fb7981e02d82a0e86a5856fc01e1b/modin/engines/ray/pandas_on_ray/io.py#L21 

However, there is an argument to be made to keeping the `PandasOnRay` portion within the name. If we decide to go in this direction, we should also update the `RayFrameManager` to `PandasOnRayFrameManager` as there may be overlaps in the future https://github.com/modin-project/modin/blob/6e24ef779c0fb7981e02d82a0e86a5856fc01e1b/modin/engines/ray/pandas_on_ray/frame/partition_manager.py#L12

**Comment by pcmoritz** [comment]
The naming looks good (but the PR description is a little out of date) :)

**Pull request merged**